src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs

```diff
@@ -1276,6 +1271,71 @@ private static StackValue[] GenerateAlphabetRange(StackValue svStart, StackValue
 
             return letters;
         }
+
+        private static StackValue[] GenerateAlphabetSequence(StackValue svStart, StackValue svEnd, StackValue svStep, StackValue svOp, RuntimeCore runtimeCore)
+        {
+            if (!svStart.IsString || !svEnd.IsInteger || !svStep.IsInteger)
+            {
+                runtimeCore.RuntimeStatus.LogWarning(
+                    WarningID.kInvalidArguments,
+                    Resources.kInvalidArgumentsInRangeExpression);
+                return null;
+            }
+
+            var startValue = runtimeCore.Heap.ToHeapObject<DSString>(svStart).Value;
+            var amount = Convert.ToInt32(svEnd.RawIntValue);
+            var step = Convert.ToInt32(svStep.RawIntValue);
+
+            // Start and end values can be just alphabet letters. So their length can't be more than 1.
+            if (startValue.Length != 1)
+            {
+                runtimeCore.RuntimeStatus.LogWarning(
+                    WarningID.kInvalidArguments,
+                    Resources.kInvalidArgumentsInRangeExpression);
+                return null;
+            }
+
+            if (amount < 0)
+            {
+                runtimeCore.RuntimeStatus.LogWarning(
+                   WarningID.kInvalidArguments,
+                   Resources.kInvalidAmountInRangeExpression);
+                return null;
+            }
+
+            if (step <= 0)
+            {
+                runtimeCore.RuntimeStatus.LogWarning(
+                   WarningID.kInvalidArguments,
+                   Resources.kRangeExpressionWithNegativeStepNumber);
+                return null;
+            }
+
+            var startLetter = startValue.ToCharArray().First();
+
+            // Alphabet sequence can be made just from letters (that are not unicode).
+            if (!Char.IsLetter(startLetter) || startLetter > 255)
+            {
+                runtimeCore.RuntimeStatus.LogWarning(
+                    WarningID.kInvalidArguments,
+                    Resources.kInvalidArgumentsInRangeExpression);
+                return null;
+            }
+
+            StackValue[] letters;
+
+            letters = Enumerable.Range(1, amount)
+                // Generate arithmetic progression.
+                 .Select(x => startLetter + (x - 1) * step)
```

Comment:
Please consider replacing this with a for loop, it should make the logic considerably clearer, and give that we're suppressing the lazy eval using toArray will be as efficient.